File: src/main/java/com/example/dpop/orchestrator/api/v1/identification/fsc/FscStep.java
package com.example.dpop.orchestrator.api.v1.identification.fsc;

import java.util.List;

public sealed interface FscStep {
    record NeedInput(List<String> missingFields) implements FscStep {}
    record Verify(String kvnr, String fsc)       implements FscStep {}
}

File: src/main/kotlin/com/example/dpop/orchestrator/api/v1/identification/fsc/FscStep.kt
package com.example.dpop.orchestrator.api.v1.identification.fsc

sealed interface FscStep {
    @JvmRecord
    data class NeedInput(val missingFields: List<String>) : FscStep

    @JvmRecord
    data class Verify(val kvnr: String?, val fsc: String?) : FscStep
}
